### ISSUE TITLE
'o vs O' corrections, ACC

### ISSUE THREAD

**Issue by drdhaval2785** [sanskrit-lexicon/CORRECTIONS#138]
After a long time, I am back for a while in the business of correction submission.
I intend to examine the http://drdhaval2785.github.io/o_vs_O/output2/ACC.html words with highest probability.
I intend to complete the issue in one part only, because there are only 58 such entries.

**Comment by drdhaval2785** [comment]
873	aparAka	apAraka

No change.

Different meanings

**Comment by drdhaval2785** [comment]
3212	kanakAvali	kanakAvalI
NO CHANGE

Avali and AvalI are synonymous.
![capture](https://cloud.githubusercontent.com/assets/5755132/10944363/d84e6368-833e-11e5-86be-7720751481d0.png)

**Comment by drdhaval2785** [comment]
3440	kalkalAya	kalkAlaya

NO CHANGE

![capture](https://cloud.githubusercontent.com/assets/5755132/10944419/184ddcdc-833f-11e5-87ed-273de01b8820.png)
![capture](https://cloud.githubusercontent.com/assets/5755132/10944430/27a5da4a-833f-11e5-990f-5db5ec2d55bc.png)

Different meanings.

**Comment by drdhaval2785** [comment]
3477	kalpavalli	kalpavallI
NO CHANGE

![capture](https://cloud.githubusercontent.com/assets/5755132/10944460/556e7ed2-833f-11e5-8f1f-9abf2268bbe9.png)

**Comment by drdhaval2785** [comment]
3976	kAlamArtaRqa	kAlamArtARqa
NO CHANGE

Both MW and PW have an entry of (uncommon) मार्ताण्ड.
So the entry कालमार्ताण्ड is justified in those dictionaries.

![capture](https://cloud.githubusercontent.com/assets/5755132/10944577/e7b3e426-833f-11e5-93ba-13115adfc483.png)

**Comment by drdhaval2785** [comment]
4673	kfzRaguru	kfzRAguru

NO CHANGE

The first is name of an author, second a type of wood.

**Comment by drdhaval2785** [comment]
42783	kfzRaya	kfzRAya

NO CHANGE

It is rather odd name, but the alphabetical order is maintained. 
![capture](https://cloud.githubusercontent.com/assets/5755132/10944660/5547842a-8340-11e5-88eb-1a0eb943d007.png)

**Comment by drdhaval2785** [comment]
6220	goRIputra 	gOrIputra
NO CHANGE

false positive.

`Note the false inclusion of 'r' and 'R'.`
@funderburkjim This is very interesting part. We are using similarity of Capital and small letters in SLP1.
And that's why such non relevant गोणीपुत्र / गौरीपुत्र issue is being enumerated. No scribe would have done such a mistake. So, these entries should be put in low priority.
But at the same time, there starts a thought that the input transliteration  for different dictionaries need an examination. If there are such cases there, we need to incorporate it.

**Comment by drdhaval2785** [comment]
6367	gopIka	gOpika
NO CHANGE

Name of a person.

**Comment by drdhaval2785** [comment]
7783	janmadIpa	janmADipa
NO CHANGE

False positive

**Comment by drdhaval2785** [comment]
7939	jarara	jAraRa
NO CHANGE

False positive

**Comment by drdhaval2785** [comment]
8052	jAtisaMkaryavAda ->	jAtisAMkaryavAda

OCR Error

![capture](https://cloud.githubusercontent.com/assets/5755132/10944975/b2710058-8341-11e5-9a66-d91a18194a55.png)

**Comment by drdhaval2785** [comment]
8777	taruRi	taruRI
NO CHANGE

The author himself has put a question mark there. We should preserve this form. The same may have been there in the catalogue of Peters.

![capture](https://cloud.githubusercontent.com/assets/5755132/10945004/e1b77bf8-8341-11e5-8b06-bf26ae196db2.png)

**Comment by drdhaval2785** [comment]
9692	darSapOrRamAsika	dArSapOrRamAsika
NO CHANGE

The author has kept `darSa` all across. Alphabetic order is preserved.
No change needed.

**Comment by drdhaval2785** [comment]
10453	draviqAH	drAviqaH
NO CHANGE

**Comment by drdhaval2785** [comment]
15909	BAnujidIkzita	BAnujIdIkzita
NO CHANGE

Even though the later reading `jI` seems more befitting to the present Hindi speaking people, the author names usually had `ji`.
See 
![capture](https://cloud.githubusercontent.com/assets/5755132/10945172/b6c332ec-8342-11e5-9c70-3ff758a92f08.png)

**Comment by drdhaval2785** [comment]
45395	maNgalavrata	maNgalAvrata
NO CHANGE

**Comment by drdhaval2785** [comment]
35873	meDAkara	meDAkAra
NO CHANGE

First a name, second - causing intelligence

**Comment by drdhaval2785** [comment]
36615	laGukaraRa	laGUkaraRa
NO CHANGE

First name of a work, second an act.

**Comment by drdhaval2785** [comment]
22171	vijjakA	vijjAkA
NO CHANGE

Alternate forms.
See
![capture](https://cloud.githubusercontent.com/assets/5755132/10945532/d4f0edee-8343-11e5-838b-a21671091db1.png)

**Comment by gasyoun** [comment]
So many no changes could mean only that there are issues with the dictionaries with which words are compared?

**Comment by drdhaval2785** [comment]
No. It is because names of works / persons can be non conventional also.

**Comment by drdhaval2785** [comment]
24980	SAlihotramuni	SAlIhotramuni
NO CHANGE

Author of MW meant `I` to be a w.r. form.
See
![capture](https://cloud.githubusercontent.com/assets/5755132/10945771/f20509f0-8344-11e5-8639-4b0ebf7e1c4d.png)

**Comment by drdhaval2785** [comment]
27287	samitsamAropaRaprAtarOpasanaprayoga	samitsamAropaRaprAtarOpAsanapAsanaprayoga
Print Error

`Opasana` is not a valid word.
`OpAsana` is.

**Comment by drdhaval2785** [comment]
38739	suDADara	suDADAra
NO CHANGE

suDA+Dara / suDA+ADAra
Both valid ones.

**Comment by drdhaval2785** [comment]
39235	hemavatI	hEmavatI
NO CHANGE

Both valid words.

**Comment by drdhaval2785** [comment]
85	agnizwomahOtra	agnizwomahotra
Not sure

both hOtra and hotra are valid words according to PWG.

**Comment by drdhaval2785** [comment]
31852	uddeSyaviDeyaboDasTAlIyavicAra ->	uddeSyaviDeyaboDasTalIyavicAra
Print Error

In the present context `sTalIya` makes more sense than `sTAlIya`.

**Comment by drdhaval2785** [comment]
3102	OrDvadehikakalpavallI	OrDvadehikakalpavalli
NO CHANGE

vallI and valli are the same.

**Comment by drdhaval2785** [comment]
3867	kArikAH	kaRikaH
NO CHANGE

False positive

**Comment by drdhaval2785** [comment]
42513	kAmasAra	kAmaSara
NO CHANGE

False positive

**Comment by drdhaval2785** [comment]
3999	kAlApAH	kAlApaH
NO CHANGES

Singular - plural issue

**Comment by drdhaval2785** [comment]
4480	kUka	kuka
NO CHANGE

Different words

**Comment by drdhaval2785** [comment]
42809	kfzRIya	kfzRiya
NO CHANGE

Different words

**Comment by drdhaval2785** [comment]
5078	kokilahOtra	kokilahotra
Not sure

Same as above 'hotra' / 'hOtra' confusion

**Comment by drdhaval2785** [comment]
5985	gIrvARapadamaYjarI	gIrvARapadamaYjari
NO CHANGE

maYjarI and maYjari both are valid words

**Comment by drdhaval2785** [comment]
6367	gopIka	gopika
NO CHANGE

Discussed ago

**Comment by drdhaval2785** [comment]
7170	camatkAraRa	camatkaraRa
NO CHANGE

Grammatically difficult to defend, but the alphabetic arrangement mandates that we keep it as it is.
![capture](https://cloud.githubusercontent.com/assets/5755132/10946398/17a6ebee-8348-11e5-9d85-5c482fd56e33.png)

**Comment by drdhaval2785** [comment]
7939	jarara	jaraRa
NO CHANGE

discussed earlier

**Comment by drdhaval2785** [comment]
8953	tArAnATa	tAranATa
NO CHANGE

Different authors

**Comment by drdhaval2785** [comment]
9251	triMSadyogAvalI	triMSadyogAvali
NO CHANGE

vallI = valli

**Comment by drdhaval2785** [comment]
10095	durgAwIkA	durgawIkA
NO CHANGE

Different words durga - durgA

**Comment by drdhaval2785** [comment]
10453	draviqAH	draviqaH
NO CHANGE

singular plural

**Comment by drdhaval2785** [comment]
12897	padyAvalI	padyAvali
NO CHANGE

AvalI = Avali

**Comment by drdhaval2785** [comment]
34454	parAtantra	paratantra
NO CHANGE

The first is name of a tantra, second - slave

**Comment by drdhaval2785** [comment]
15338	brahmanAmAvalI	brahmanAmAvali
NO CHANGE

AvalI = Avali

**Comment by drdhaval2785** [comment]
35349	BEkzAcaraRa	BEkzacaraRa
NO CHANGE

Different words

**Comment by drdhaval2785** [comment]
18520	yavanAH	yavanaH
NO CHANGE

singular plural

**Comment by drdhaval2785** [comment]
20224	rAmavijYApanAstotra	rAmavijYApanastotra
NO CHANGE

Both valid words

**Comment by drdhaval2785** [comment]
21283	vaCAcArya	vacAcArya
NO CHANGE

Alternative words

See MW
![capture](https://cloud.githubusercontent.com/assets/5755132/10946655/7acec02e-8349-11e5-9a9a-6109bde15f29.png)

**Comment by drdhaval2785** [comment]
22484	viBrAzwezwiprayoga	viBrazwezwiprayoga
Not sure.

I incline towards correction from `viBrAzw` to `viBrazw`
See MW
![capture](https://cloud.githubusercontent.com/assets/5755132/10946711/cc3dd56c-8349-11e5-8bf7-e1d516c6ca3b.png)

**Comment by drdhaval2785** [comment]
23376	veNkawaBEw	veNkawaBew
NO CHANGE

ACC gives two other similar names also ending with `bhEw`
![capture](https://cloud.githubusercontent.com/assets/5755132/10946821/6386da0e-834a-11e5-868b-6c211fc379b6.png)

**Comment by drdhaval2785** [comment]
24901	SAMBavAnandakalpa	SAmbavAnandakalpa
NO CHANGE

**Comment by drdhaval2785** [comment]
37742	SiNgABawwa	SiNgaBawwa
Not sure.

I incline towards correction.
ACC
![capture](https://cloud.githubusercontent.com/assets/5755132/10946887/ba8c3664-834a-11e5-8aea-a4493a8fa500.png)
MW
![capture](https://cloud.githubusercontent.com/assets/5755132/10946894/cebc6816-834a-11e5-98d6-1784aae0d28d.png)

**Comment by drdhaval2785** [comment]
26387	zawpaYcASatikAhorA	zawpaYcASatikahorA
NO CHANGE

Name of a work.
Grammatically defensible.

**Comment by drdhaval2785** [comment]
26475	zoqaSanityAtantra	-> zoqaSanityatantra
Print error

`nityAtantra` doesn't make sense.

**Comment by drdhaval2785** [comment]
26544	saMskArAH	saMskAraH
NO CHANGE

singular plural

**Comment by drdhaval2785** [comment]
27784	sAmAnyalakzaRAvyaBicAra	sAmAnyalakzaRavyaBicAra
NO CHANGE

'lakzaRA' and 'lakzaRa' both grammatically defensible forms

**Comment by drdhaval2785** [comment]
48082	haMsapArameSvara	haMsaparameSvara
NO CHANGE

Name of a work. Grammatically defensible

**Comment by drdhaval2785** [comment]
This ends the highest probability entries (58 in toto).
Now it is time for @zaaf2 to review and @funderburkjim to install.

The reason behind taking up ACC was to be able to complete it in one go (58 entries).

I would be working sporadically in smaller dictionaries. Not enough time to undertake majors.

**Comment by gasyoun** [comment]
@drdhaval2785 quick and documented just enough, to be quick - that matters.

ACC is important in that matter, that I assume that most words gone wrong now in headwords are titles of books, authors or place names. So this is where a clean ACC is even more important than clean MW, because it is the only other source in many cases.
Actually the not so high possibility cases contain not less gems, as I see it now:

* 136	11227	navagrahayantrodDAraRakrama	navagrahayantrodDaraRakrama
* 154	15826	BAgavatatattvadIpaprakASAvaraRaBaNga	BAgavatatattvadipaprakASAvaraRaBaNga
* 183	27779	sAmAnyalakzaRApUrvapakzaprakASa	sAmAnyalakzaRapUrvapakzaprakASa

Why are they important? Out of 430k words in Reverse dictionary 3k are long. If we have 500 long words that should not be there, because of a convention used by ACC, not used by MW, would be no reason to leave both.

* 991	26700	saMKyAmuzwyaDikaraRAkzepa	saMKyAmuzwyaDikaraRakzepa
* 912	21335	vanaBojanapuRyAhavAcanaprayoga	vanaBojanapuRyAhavacanaprayoga
* 778	14042	pratiyogijYAnakAraRatAvAda	pratiyogijYAnakaraRatAvAda

**Comment by zaaf2** [comment]
re. `22484` viBrAzwezwiprayoga viBrazwezwiprayoga
I agree. There is no विभ्राष्ट.
MW: विभ्रष्ट [p= 979] : mfn. fallen, sunk.

re. `37742` SiNgABawwa SiNgaBawwa
It seems the author considered SiNgABawwa as a valid form. He gives the name of SiNgaBawwa’s work as शिङ्गभट्टीय (as in MW), but quotes another source with ā:
![image](https://cloud.githubusercontent.com/assets/14730665/11005278/adecff9a-849b-11e5-8a7a-60aefce05003.png)

**Comment by zaaf2** [comment]
I have reviewed the other cases and I found nothing to add.

**Comment by zaaf2** [comment]
@drdhaval2785 Good job! High quality work ― and fast!

**Comment by gasyoun** [comment]
Fast as devil.

**Comment by funderburkjim** [comment]
re `Why were vijjA and vijjAkA not shown in ACC dictionary ? Missing alternate headwords?`

Right. Only in MW has an attempt been made to deal with alternate headwords.  

See #133 , where some preliminary remarks appear relating to the question of alternate headwords.

**Comment by funderburkjim** [comment]
My reading of these leads me to harvest these 4 corrections.  
```
27287 samitsamAropaRaprAtarOpasanaprayoga -> samitsamAropaRaprAtarOpAsanapAsanaprayoga : print error 
31852 uddeSyaviDeyaboDasTAlIyavicAra -> uddeSyaviDeyaboDasTalIyavicAra : print error
26475 zoqaSanityAtantra -> zoqaSanityatantra : Print error
22484 viBrAzwezwiprayoga ->  viBrazwezwiprayoga : Print error
```

**Comment by funderburkjim** [comment]
re `zoqaSanityAtantra -> zoqaSanityatantra`

Changed spelling in two listings (26475 and 47496)
And also under headword kAdimata (39592)

Also there is a feminine form nityA in MW.

**Comment by funderburkjim** [comment]
Changes installed.

**Comment by gasyoun** [comment]
After doing for the 2nd time something that has already been "fixed" in another topics, but not yet integrated, I'm thinking again about the process. The steps are as:
1) Anyone proposes a change
2) Jim agrees
3) Jim implements
4) Jim regenerates sanhw
5) Dhaval regenerates Ovso
First three are out my reach. The question is if we should and how much time would it take to have  weekly or bi-weekly update. To have one Python that does 2 last ones in 1 click - possible, @funderburkjim or not needed?

**Comment by funderburkjim** [comment]
@gasyoun   Regarding automation of items 4/5:   It would take, probably, a few hours from me and Dhaval to add more automation to these steps.  Weighed against one duplicated effort ,  the balance probably does not yet warrant the effort.  It's reasonable to monitor this, and if the duplicated effort gets large,  then it would be worth the automation effort.

**Comment by gasyoun** [comment]
Agree.

**Comment by funderburkjim** [comment]
Incidentally, there is a need for further automation in those steps 2-4 that currently only I can do. For instance, in processing the ACC corrections yesterday,  it took me about 3 hours, probably about as long as it took Dhaval to generate his analyses.  

Let me try to describe what's involved in these steps that are currently hidden from others, in the context of these ACC changes.  

step1. I open up a 'notes.txt' file for the corrections in this issue.   

step2. I read through the issue comments, one by one.  For each one I try to understand what the comment-maker (Dhaval in this case) is saying.  If I understand the comment and agree with it, then 
  *  If it is a NO CHANGE,  I do nothing more
  *  If it is a correction, I make a brief note of the form 'case#  old-spelling -> new-spelling : reason'.
    For instance '26475 zoqaSanityAtantra -> zoqaSanityatantra : Print error'

step3. Once notes.txt is finished, then the installation process begins.

step4. Each of the changes indicated in notes.txt involves generating one (or sometimes more than one) change transaction for the acc.txt base digitization.  In practice, then, I must generate a file, let's call it accupd138.txt, containing the transactions implied by notes.txt.   Here's what one of the transactions for the above mentioned print error looks like:
```
63604 old <HI>{#zoqaSanityAtantra#}¦ See Ka1dimata.
63604 new <HI>{#zoqaSanityatantra#}¦ See Ka1dimata.
```
   The transaction consists of two lines.  The first line contains the line-number within the acc.txt file,
   the keyword 'old', and the text of that line 63604 of acc.txt.  The second line is similar, and contains the correction.

 In the case of this batch of ACC changes, since there were so few of them, I generated the transactions in the accupd138.txt file manually, That is, I edited a local copy of the acc.txt file,  located the string 'zoqaSanityAtantra' , noted the line number, copied into the clipboard the text,
and constructed the two lines by typing and pasting, then correted the second line.   In this case,
my search for the string zoqaSanityAtantra led to five hits, and I decided that they should also be changed and similarly generated change transactions for the other four in accupd138.txt.

 If there were sufficiently more changes in some other batch (more than the 4 of notes.txt in this case), I would have adapted a program to help do part of what I did manually in generating the transactions.

step5. Once accupd138.txt is finished,  the actual installation of the changes is well-automated.  I typically append the current batch of transactions to the latest 'pool' of transactions; in this case this transaction pool is a file named manualByLine_slp1.txt in the pywork directory for dictionary 'acc'.
Then I run a python script to generate a new edition of the latest version of the acc digitization as file acc.txt.  Another bash script (redo_hw.sh)  regenerates the headword list (acchw2.txt) from acc.txt.  Finally, another bash script (redo_xml.sh) regenerates the xml version of acc (acc.xml), and the sqlite version of acc (acc.sqlite) that is used in the displays for acc.

step6.  Now, several documentation steps are done.
* change the date in the downloads page for acc,
* redo the download scripts, to regenerate the download zip files
* generate correction records for the 'CORRECTIONS'  Github repository. This 'generation' step is usually done in parallel with step4 above.  In this case, I did it by bringing up an ACC display, looking up each word (such as zoqaSanityAtantra), clicking the 'Corrections' button, and filling in the form. When a correction form is submitted, that submission updates a particular file 'cfr.tsv' from the form.
* install the CORRECTIONS records in the Github repository.  This involves:
  * modifying cfr.tsv to indicate that corrections have been made (i.e., that step5 has been completed for these acc corrections)
  * downloading the cfr.tsv file to my local copy of the CORRECTIONS repository
  * running a local python script to update the 'correctionform.txt' file of the repository
  * manually updating the history.txt file of the repository
  * manually updating the dictionaries/acc/acc_printchange.txt file
  * manually syncing the local revised CORRECTIONS repository to the official repository at GitHub

step7.  Now come post-processing steps, such as regeneration of sanhw1/2.   Typically (as with ACC) I don't regularly do these steps.  For instance, for sanhw1/2 regeneration,  a script is run on Cologne server to generate the sanhw1.txt file,  this file is downloaded and installed into the local CORRECTIONS repository, and a commit-sync step brings it into availability on Github.  There are
other post-processing steps also.

But that's enough detail. Surely the reader of this description is becoming glassy-eyed.  

There must be a better way!  that reader is no doubt saying.
